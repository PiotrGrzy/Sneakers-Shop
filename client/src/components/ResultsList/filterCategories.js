// cant filter by  multiple choice parameter in graphCMS query

const filterCategories = (data, categories) => {
  return data.filter(item =>
    item.category.some(cat => categories.includes(cat))
  );
};

export default filterCategories;
