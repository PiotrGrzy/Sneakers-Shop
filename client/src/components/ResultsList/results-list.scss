.results-list {
  max-width: 1400px;
  margin: 0 auto;
  margin-top: 2rem;
  padding: 0 4rem;
  justify-content: center;
  width: 100%;
  display: grid;
  grid-template-columns: repeat(auto-fit, 30rem);
  gap: 1rem;
}
