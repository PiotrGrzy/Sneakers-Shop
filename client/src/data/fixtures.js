export const item = {
  id: 'ck75wcleo155b0b843rlkt044',
  price: 129,
  gender: 'Men',
  category: ['Running'],
  imageMain: {
    id: 'ck75w55qoe0v90b201x2oqo8l',
    url: 'https://media.graphcms.com/JqjJAtRTlethzLQxhUtQ',
  },
  model: 'Zoom 210',
  imageSecondary: {
    id: 'ck75w55kfe0v40b209n4moxvp',
    url: 'https://media.graphcms.com/EVQOXidUTZksHn7HzJAQ',
  },
  sizes: '40 41 42 43 44 45 46 47',
  brand: 'Nike',
  trending: true,
};
