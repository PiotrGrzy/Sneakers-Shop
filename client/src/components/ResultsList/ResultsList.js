import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import SneakerLI from '../SneakerLI/SneakerLI';
import { selectSearchResults } from '../../redux/shop/shop.selectors';
import { setLoading } from '../../redux/shop/shop.actions';
import './results-list.scss';
import Spinner from '../spinner/Spinner';
import filterCategories from './filterCategories';

export const ResultsList = ({ searchResults, isLoading, searchQuery }) => {
  if (isLoading)
    return (
      <div className="results-list">
        <Spinner />
      </div>
    );

  return (
    <div className="results-list">
      {searchResults.length < 1 ? (
        <p className="results-list__noresults">
          No results with given criteria
        </p>
      ) : searchQuery.category.length > 0 ? (
        // instant filtering on frontend, this should not be filtered in this component, todo: fix it
        filterCategories(searchResults, searchQuery.category).map((item) => (
          <SneakerLI key={item.id} sneaker={item} />
        ))
      ) : (
        searchResults.map((item) => <SneakerLI key={item.id} sneaker={item} />)
      )}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    searchResults: selectSearchResults(state),
    searchQuery: state.search,
    isLoading: state.shop.isLoading,
  };
};

export default connect(mapStateToProps, { setLoading })(ResultsList);
