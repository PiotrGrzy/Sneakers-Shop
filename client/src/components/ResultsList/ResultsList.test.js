import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { ResultsList } from './ResultsList';
import Spinner from '../spinner/Spinner';
import { items } from '../../data/fixtures';

const props = {
  searchResults: items,
  isLoading: false,
  searchQuery: {
    category: [],
  },
};

describe('ResultsList', () => {
  const wrapper = shallow(<ResultsList {...props} />);
  console.log(wrapper.debug());
  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders proper num of items', () => {
    expect(wrapper.find('SneakerLi').length).toEqual(3);
  });

  it('renders information if no search results', () => {
    const wrapper2 = shallow(<ResultsList {...props} searchResults={[]} />);
    expect(wrapper2.find('.results-list__noresults').text()).toEqual(
      'No results with given criteria'
    );
  });
});
