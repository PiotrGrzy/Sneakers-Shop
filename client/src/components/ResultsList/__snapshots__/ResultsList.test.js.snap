// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList renders properly 1`] = `ShallowWrapper {}`;
